### PR TITLE
🔧 #161: Refined the chat stream req/responses

### PR DESCRIPTION
**Pull request by roma-glushko**
## Changes

- Separated chat & chat stream request schemas 
- introduced a new finish reason field 
- added metadata to stream chat response
- allow to attach some metadata to a chat stream request and then attach it to each chat stream chunk
- adjusted error message schema to include request ID and metadata

### CODE REVIEW

File: pkg/api/schemas/chat.go [hunk 1]
```diff
@@ -17,7 +17,7 @@ func NewChatFromStr(message string) *ChatRequest {
 		Message: ChatMessage{
 			"human",
```
Comment (mkrueger12) [inline pkg/api/schemas/chat.go:18]: I think this needs to be "user" instead of "human"

Comment (mkrueger12) [reply pkg/api/schemas/chat.go:18]: Since all providers take "user" as input in request schema

File: pkg/api/schemas/chat_stream.go [hunk 2]
```diff
@@ -1,31 +1,55 @@
...
+	Message        ChatMessage          `json:"message" validate:"required"`
+	MessageHistory []ChatMessage        `json:"messageHistory" validate:"required"`
+	Override       *OverrideChatRequest `json:"overrideMessage,omitempty"`
+	Metadata       *Metadata            `json:"metadata,omitempty"`
+}
+
+func NewChatStreamFromStr(message string) *ChatStreamRequest {
+	return &ChatStreamRequest{
+		Message: ChatMessage{
+			"human",
```
Comment (mkrueger12) [inline pkg/api/schemas/chat_stream.go:22]: Should be "user" instead of human"

**Review by mkrueger12 — APPROVED** [review]

File: pkg/api/schemas/chat.go (hunk 1 above)
Comment (roma-glushko) [reply pkg/api/schemas/chat.go:18]: I see yeah it makes sense to change that. However, it makes almost no effect as it's used in tests only to mock request schema. I will take care of this in a followup PR 👀

**Pull request merged**